packages/cloudflare/src/utils/generate-routes-json.ts

```diff
@@ -0,0 +1,267 @@
+import type { AstroConfig, RouteData, RoutePart } from 'astro';
+
+import { existsSync } from 'node:fs';
+import { writeFile } from 'node:fs/promises';
+import { posix } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import glob from 'tiny-glob';
+import { removeLeadingForwardSlash, removeTrailingForwardSlash } from './assets.js';
+
+function prependForwardSlash(path: string) {
+	return path[0] === '/' ? path : `/${path}`;
+}
+
+export const getParts = (part: string) => {
+	const result: RoutePart[] = [];
+	part.split(/\[(.+?\(.+?\)|.+?)\]/).map((str, i) => {
+		if (!str) return;
+		const dynamic = i % 2 === 1;
+
+		const [, content] = dynamic ? /([^(]+)$/.exec(str) || [null, null] : [null, str];
+
+		if (!content || (dynamic && !/^(?:\.\.\.)?[\w$]+$/.test(content))) {
+			throw new Error('Parameter name must match /^[a-zA-Z0-9_$]+$/');
+		}
+
+		result.push({
+			content,
+			dynamic,
+			spread: dynamic && /^\.{3}.+$/.test(content),
```

Comment:
Don't put regex inside loops; it's really bad for performance. At least define them inside the function before the loop.

Reply:
I copied this from https://github.com/withastro/astro/blob/3776ecf0aa9e08a992d3ae76e90682fd04093721/packages/astro/src/core/routing/manifest/create.ts#L45-L70, because I wanted to reuse it, but it isn't exported from Astro. 

I would prefer to have it exported from Astro. I could also import it from `node_modules` manually, but that is also not great.

Anyways I did copied it before it was changed in Astro, so my version is already outdated, so for now I'll copy the new version.